### PR TITLE
Cleanup IR v7 from the MO

### PR DESCRIPTION
**Pull request by lazarevevgeny**
Description: Remove support of IRv7 in the Model Optimizer

JIRA: CVS-33781, CVS-33782

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR: N/A no new transformations enabled
* [x]  Transformation preserves original framework node names: N/A no new transformations enabled

Validation:
* [x]  Unit tests: unit tests updated
* [x]  Framework operation tests: N/A no new operations enabled
* [x]  Transformation tests: N/A no new transformations enabled
* [x]  e2e model test with an update of ./tests/e2e_oss/utils/reshape_utils.py: N/A no new models enabled
* [x]  Model Optimizer IR Reader check: N/A no new modes enabled

Documentation:
* [x]  Supported frameworks operations list: N/A no new operations enabled
* [x]  Supported **public** models list: N/A no new models enabled
* [x]  New operations specification: N/A no new operations enabled
* [x]  Guide on how to convert the **public** model: N/A no new models enabled
* [x]  User guide update: CVS-33785

### CODE REVIEW

File: model-optimizer/extensions/back/blob_normalizer.py [hunk 1]
```diff
@@ -36,8 +36,7 @@ class BlobNormalizer(BackReplacementPattern):
     enabled = True
 
     graph_condition = [
-        lambda graph: graph.graph['cmd_params'].blobs_as_inputs or
-                      graph.graph['cmd_params'].generate_experimental_IR_V10
+        lambda graph: graph.graph['cmd_params'].blobs_as_inputs
```
Comment (jane-intel) [inline model-optimizer/extensions/back/blob_normalizer.py:39]: We should delete blobs_as_inputs as well. IRV10 can not contain blobs. The question is -- if we would like to do it here.

File: model-optimizer/extensions/ops/psroipooling.py [hunk 2]
```diff
@@ -47,10 +47,10 @@ def supported_attrs(self):
             'spatial_bins_y',
             'pooled_width',
             'pooled_height',
+            'no_trans',
```
Comment (jane-intel) [inline model-optimizer/extensions/ops/psroipooling.py:50]: Please delete -- it was for v7 only

File: model-optimizer/mo/middle/passes/fusing/fuse_linear_ops.py [hunk 3]
```diff
@@ -254,19 +247,10 @@ def fuse_linear_ops(graph: Graph):
...
             is_fused = _fuse_mul(graph, node, fuse_nodes, False)
 
-        # Fuse Add to Convolution/FC
-        if hasattr(graph, 'graph') and 'cmd_params' in graph.graph and \
-                not graph.graph['cmd_params'].generate_experimental_IR_V10:
-            if node.soft_get('op') == 'Add' and \
-                    get_value_in_port(node) is not None and \
-                    node.soft_get('can_be_fused') is True:
-                fuse_nodes = forward_bfs(node, [], ['MatMul'])
-                is_fused = _fuse_add(graph, node, fuse_nodes, False)
```
Comment (jane-intel) [inline model-optimizer/mo/middle/passes/fusing/fuse_linear_ops.py:268]: It seems `_fuse_add` is not used, if so lets remove the function.

File: model-optimizer/mo/utils/cli_parser.py [hunk 4]
```diff
@@ -312,9 +314,8 @@ def get_common_cli_parser(parser: argparse.ArgumentParser = None):
                                    'Use --input option to specify a value for freezing.',
                               default=None)
     common_group.add_argument('--generate_deprecated_IR_V7',
-                              help='Force to generate old deprecated IR V7'
-                                   ' with layers from old IR specification.',
-                              action=DeprecatedStoreTrue,
+                              help='Force to generate deprecated IR V7 with layers from old IR specification.',
+                              action=IgnoredAction,
```
Comment (jane-intel) [inline model-optimizer/mo/utils/cli_parser.py:318]: Should we mark V10 the same?

**Comment by lazarevevgeny** [comment]
> This is amazing, but please address the comments below
> 
> And... does IR Reader check run in the CI? I think that potentially we may have affected it. Paranoiac but still valid question.
> 
> Were there some nightly runs triggered? Performance checks? Benchmark? It is a huge change. I am just curious -- no pressure here.

I have passed benchmark run with just one new failure (DLRM model and it looks like an infrastructure bug because the error message states that the size of the input data is wrong). You can find the results here "try/job/ubuntu18_i7_8700/1141"

File: model-optimizer/extensions/front/disable_weights_quantize_value_propagation.py [hunk 5]
```diff
@@ -22,7 +22,6 @@ class DisableQuantizeValuePropagation(FrontReplacementPattern):
     enabled = True
     graph_condition = [
         lambda graph: graph.graph['cmd_params'].keep_quantize_ops_in_IR
```
Comment (jane-intel) [inline model-optimizer/extensions/front/disable_weights_quantize_value_propagation.py:24]: If we remove blobs_as_inputs, then keep_quantize_ops_in_IR should go too, as we do not have statistics section in V10 IR structure.

**Pull request merged**